chain_bridge.go

```diff
@@ -200,6 +200,37 @@ func (l *LndRpcChainBridge) EstimateFee(ctx context.Context,
 	return l.lnd.WalletKit.EstimateFeeRate(ctx, int32(confTarget))
 }
 
+// SubscribeCustomMessages creates a subscription to custom messages received
+// from our peers.
+//
+// TODO(ffranr): This should be moved into its own interface because it's not
+// related to the chain.
+func (l *LndRpcChainBridge) SubscribeCustomMessages(
+	ctx context.Context) (<-chan lndclient.CustomMessage,
+	<-chan error, error) {
+
+	return l.lnd.Client.SubscribeCustomMessages(ctx)
+}
+
+// SendCustomMessage sends a custom message to a peer.
+//
+// TODO(ffranr): This should be moved into its own interface because it's not
+// related to the chain.
+func (l *LndRpcChainBridge) SendCustomMessage(ctx context.Context,
+	msg lndclient.CustomMessage) error {
+
+	return l.lnd.Client.SendCustomMessage(ctx, msg)
+}
+
+// InterceptHtlcs intercepts all incoming HTLCs and calls the given handler
+// function with the HTLC details. The handler function can then decide whether
+// to accept or reject the HTLC.
+func (l *LndRpcChainBridge) InterceptHtlcs(
```

Comment:
This could be move to a like `HtlcSwapControl` interface or something like that.